src/Raven.Server/Documents/Sharding/Handlers/Processors/Streaming/ShardedStreamingHandlerProcessorForGetStreamQuery.cs

```diff
@@ -0,0 +1,156 @@
+        {
+            //TODO stav: use the same time limited token for the sharded operation (as the non sharded)?
+            using (context.OpenReadTransaction())
+            {
+                var queryProcessor = new ShardedQueryProcessor(context, RequestHandler, query);
+                if (queryProcessor.IsMapReduce())
+                    throw new NotSupportedInShardingException("MapReduce is not supported in sharded streaming queries");
+
+                //TODO stav: do we support includes in sharded streaming?
+                queryProcessor.Initialize();
+
+                //TODO stav: Initialize() rewrites the query for paging but creates a blittable of it inside the processor and doesn't modify this 'query' var by ref
+
+                var cmds = new PostQueryStreamCommand[RequestHandler.DatabaseContext.ShardCount];
+                for (int i = 0; i < cmds.Length; i++)
+                {
+                    cmds[i] = new PostQueryStreamCommand(query.ToJson(context), format, debug, ignoreLimit, properties); //TODO stav: query.ToJson(context) needs to be queryTemplate from the processor!
+                }
+
+                //TODO stav: continuation token needs to get the offset and limit AFTER the rewrite - needs to be changed
+                var continuationToken = RequestHandler.ContinuationTokens.GetOrCreateContinuationToken(context, query.Offset ?? 0, query.Limit ?? Int32.MaxValue);
+
+                var op = new ShardedStreamQueryOperation(HttpContext, cmds);
+                var results = await RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(op);
+                using var streams = await results.Result.InitializeAsync(RequestHandler.DatabaseContext, HttpContext.RequestAborted);
+
+                //TODO stav: query metadata also needs to be after the rewrite
+                IAsyncEnumerable<BlittableJsonReaderObject> documents = OrderDocumentsByQuery(streams, query.Metadata, continuationToken);
+
+                //TODO stav: need to combine QueryStats and have them be written to stream first
+                await using (var writer = new AsyncBlittableJsonTextWriter(context, RequestHandler.ResponseBodyStream()))
+                {
+                    writer.WriteStartObject();
+
+                    await writer.WriteArrayAsync("Results", documents);
+
+                    writer.WriteEndObject();
+                }
+            }
+        }
+
+        protected override RequestTimeTracker GetTimeTracker()
+        {
+            return new RequestTimeTracker(HttpContext, Logger, null, "StreamQuery", doPerformanceHintIfTooLong: false);
+        }
+
+        protected override async ValueTask<BlittableJsonReaderObject> GetDocumentData(TransactionOperationContext context, string fromDocument)
+        {
+            var ids = ShardLocator.GetDocumentIdsByShards(context, RequestHandler.DatabaseContext, new[] {fromDocument});
```

Comment:
there could be only one document here, no?

Reply:
yes. changed